gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java

```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
     compute.delete(firstAddressId);
     compute.delete(secondAddressId);
   }
+
+  @Test
+  public void testCreateGetAndDeleteStandardDisk() throws InterruptedException {
+    String name = BASE_RESOURCE_NAME + "create-and-get-standard-disk";
+    DiskId diskId = DiskId.of(ZONE, name);
+    DiskInfo diskInfo =
+        DiskInfo.of(diskId, StandardDiskConfiguration.of(DiskTypeId.of(ZONE, "pd-ssd"), 100L));
+    Operation operation = compute.create(diskInfo);
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    // test get
+    Disk remoteDisk = compute.get(diskId);
+    assertNotNull(remoteDisk);
+    assertEquals(ZONE, remoteDisk.diskId().zone());
+    assertEquals(diskId.disk(), remoteDisk.diskId().disk());
+    assertNotNull(remoteDisk.creationTimestamp());
+    assertNotNull(remoteDisk.id());
+    assertTrue(remoteDisk.configuration() instanceof StandardDiskConfiguration);
+    assertEquals(100L, (long) remoteDisk.configuration().sizeGb());
+    assertEquals("pd-ssd", remoteDisk.configuration().diskType().diskType());
+    assertNull(remoteDisk.lastAttachTimestamp());
+    assertNull(remoteDisk.lastDetachTimestamp());
+    // test get with selected fields
+    remoteDisk = compute.get(diskId, Compute.DiskOption.fields());
+    assertNotNull(remoteDisk);
+    assertEquals(ZONE, remoteDisk.diskId().zone());
+    assertEquals(diskId.disk(), remoteDisk.diskId().disk());
+    assertNull(remoteDisk.creationTimestamp());
+    assertNull(remoteDisk.id());
+    assertTrue(remoteDisk.configuration() instanceof StandardDiskConfiguration);
+    assertNull(remoteDisk.configuration().sizeGb());
+    assertEquals("pd-ssd", remoteDisk.configuration().diskType().diskType());
+    assertEquals(DiskConfiguration.Type.STANDARD, remoteDisk.configuration().type());
+    assertNull(remoteDisk.lastAttachTimestamp());
+    assertNull(remoteDisk.lastDetachTimestamp());
+    operation = remoteDisk.delete();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    assertNull(compute.get(diskId));
+  }
+
+  @Test
+  public void testCreateGetAndDeleteImageDisk() throws InterruptedException {
+    String name = BASE_RESOURCE_NAME + "create-and-get-image-disk";
+    DiskId diskId = DiskId.of(ZONE, name);
+    DiskInfo diskInfo = DiskInfo.of(diskId, ImageDiskConfiguration.of(IMAGE_ID));
+    Operation operation = compute.create(diskInfo);
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    // test get
+    Disk remoteDisk = compute.get(diskId);
+    assertNotNull(remoteDisk);
+    assertEquals(ZONE, remoteDisk.diskId().zone());
+    assertEquals(diskId.disk(), remoteDisk.diskId().disk());
+    assertEquals(DiskInfo.CreationStatus.READY, remoteDisk.creationStatus());
+    assertNotNull(remoteDisk.creationTimestamp());
+    assertNotNull(remoteDisk.id());
+    assertTrue(remoteDisk.configuration() instanceof ImageDiskConfiguration);
```

Comment:
Nit: if you want, you could save some configuration() calls and casting if you save remoteDisk.configuration() to a variable (here and for similar cases below).